server/hotrod/src/main/java/org/infinispan/server/hotrod/Encoder2x.java

```diff
@@ -692,6 +743,11 @@ private Optional<AbstractTopologyResponse> generateTopologyResponse(short client
          }
       }
 
+      if (topologyId != 1) {
```

Comment:
not sure why this condition is necessary. Changing line https://github.com/infinispan/infinispan/blob/9348bbc6679b5e88bf08555627abce310bde5e63/server/hotrod/src/main/java/org/infinispan/server/hotrod/Encoder2x.java#L666 should be enough, right?

Reply:
Updated to clean this up a bit more, please have a look.